plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -252,7 +276,8 @@ protected void copyTableSchema(Connection connection, String catalogName, String
                 return Optional.of(timeColumnMapping(TIME));
 
             case Types.TIMESTAMP:
-                return Optional.of(timestampColumnMappingUsingSqlTimestamp(TIMESTAMP));
+                int precision = typeHandle.getDecimalDigits().orElse(SQLSERVER_MAX_SUPPORTED_TIMESTAMP_PRECISION);
```

Comment:
can decimal digits be missing? when?

Reply:
I'm not sure but what about when you use `TIMESTAMP '2021-01-20 12:26:00'` and not `TIMESTAMP(0) '2021-01-20 12:26:00'`? SQL Server would treat `DATETIME2` as `DATETIME2(7)`.

I made the assumption that decimal digits can be empty after seeing that `getRequiredDecimalDigits()` exist.

Reply:
> I made the assumption that decimal digits can be empty after seeing that getRequiredDecimalDigits() exist.

it's because they are not present for other data types (like varchar, or bigint),
but i'd expect them to be present in this case. let's use `getRequiredDecimalDigits` and fall back only if needed (and then prove the need with a test)

Reply:
You, sir, were right! The precision for timestamps is always inferred from literal values if not provided explicitly.